pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -203,6 +203,10 @@ def new_process(self, process_idx, trainer, mp_queue):
         # persist info in ddp_spawn
         self.transfer_distrib_spawn_state_on_fit_end(results)
 
+        # ensure that spawned processes go through teardown before joining (for more context, see
+        # https://github.com/PyTorchLightning/pytorch-lightning/pull/8217#issuecomment-871261949).
+        trainer._call_teardown_hook(self.lightning_module)
```

Comment:
Thanks for thinking about my comment here and including it. 

However, I think it would be better to keep a small comment here and either point to the full pull request or not put a link at all. While the contents of my comments are correct, the formulation is not very formal to be included in the source could I believe.